### ISSUE TITLE
Trailing `/` in i18n Redirects to `/learn`

### ISSUE THREAD

**Issue by ShaunSHamilton** [freeCodeCamp/freeCodeCamp#44818]
Noticed in: https://github.com/freeCodeCamp/freeCodeCamp/pull/44809

**To Reproduce**

1) Go to: https://www.freecodecamp.org/portuguese/learn/
2) See `301` redirect to `/learn`

OR

1) Go to https://www.freecodecamp.org/learn/
2) Choose another language
3) Refresh the page
4) See `301` to `/learn`

**Comment by moT01** [comment]
More not working:
https://www.freecodecamp.org/portuguese/learn/responsive-web-design/
https://www.freecodecamp.org/portuguese/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements/
https://www.freecodecamp.org/portuguese/learn/responsive-web-design/#basic-html-and-html5

Working:
https://www.freecodecamp.org/portuguese/learn/responsive-web-design
https://www.freecodecamp.org/portuguese/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements

Other than Chinese (https://freecodecamp.org/chinese/learn/ works), the rest of the languages seem to do the same.

Also, it doesn't seem like anything works for chinese-traditional - I can't get on any pages for it. They take me to `chinese.freecodecamp.org/-traditional` Pretty sure that's the case anyway.

**Comment by nhcarrigan** [comment]
> Also, it doesn't seem like anything works for chinese-traditional - I can't get on any pages for it. They take me to chinese.freecodecamp.org/-traditional Pretty sure that's the case anyway.

@ojeytonwilliams do we have a redirect that might be doing this?

**Comment by moT01** [comment]
I want to say that the `chinese` redirect is catching the `chinese-traditional` URLs

**Comment by ojeytonwilliams** [comment]
> If the proxy_pass directive is specified with a URI, then when a request is passed to the server, the part of a normalized request URI matching the location is replaced by a URI specified in the directive:
location /name/ {
    proxy_pass http://127.0.0.1/remote/;
}

This might be relevant.  What I don't understand is why https://www.freecodecamp.org/espanol/ is fine, as is https://www.freecodecamp.org/espanol, but https://www.freecodecamp.org/espanol/learn/ is not.  If it were just that /espanol/ was being replaced with /, I'd have thought that would be a problem for all espanol links, not just subpaths.

**Comment by ojeytonwilliams** [comment]
> I want to say that the `chinese` redirect is catching the `chinese-traditional` URLs. Hence, why you get the `-traditional` at the end of the URL there. Not positive though.

```
  location ~ ^/chinese/?(.*)$ {
    return 302 https://chinese.freecodecamp.org/$1;
  }
```

Yeah, that question mark is messing us up.  It makes the `/` optional, so `/chinese-traditional` matches when it shouldn't.

**Comment by KitaPDev** [comment]
@ShaunSHamilton Hi, I would like the opportunity to work on this issue as part of my university's Software Development class. Can I be assigned to work on this?

**Comment by ShaunSHamilton** [comment]
@KitaPDev Thank you, for your interest in contributing. However, this issue is not currently open for contribution.

An issue with the _help wanted_ or _first timers only_ label is open for contribution. The first comprehensive PR created will be reviewed and merged. We typically do not assign issues to anyone other than long-time contributors.

If you would like to contribute, and have not read the contributors docs, please do so here: https://contribute.freecodecamp.org/#/

If you have any issues with contributing, be sure to join us on the [contributors channel](https://chat.freecodecamp.org/channel/contributors), or on the [contributors sub-forum](https://forum.freecodecamp.org/c/contributors/3)

**Comment by raisedadead** [comment]
Did you not deploy something to fix this?

**Comment by ojeytonwilliams** [comment]
Nope, I fixed the issue where https://www.freecodecamp.org/chinese-traditional/ sent you to https://chinese.freecodecamp.org/-traditional/ 

As I understand it, this is happening because of the trailing `/` after proxy_pass in the NGINX config. 

i.e. https://github.com/freeCodeCamp/nginx-config/blob/98b8a74ec5cbc08411cd64108b8fc910a7e95c96/sites-enabled/10-www.freecodecamp.org.conf#L181-L183

This is a problem, because if the location block is `/espanol/` then the trailing slash means 'espanol' gets removed from the path proxied request. For example, if I request `https://www.freecodecamp.org/espanol/learn/` then serve sees a request to `https://www.freecodecamp.org/learn/` which isn't what we want.

The fix seems to be to 1) remove those and 2) move the current `public` directories into `public/language`

If we do that NGINX should start keeping the matching paths intact and that's fine because serve will find things in the `espanol` directory to serve.

At least that's what I think's going on!

**Comment by sidemt** [comment]
The same issue is happening for Japanese too.
